src/Simplex/Chat/Types.hs

```diff
@@ -649,6 +649,23 @@ instance ToField ImageData where toField (ImageData t) = toField t
 
 deriving newtype instance FromField ImageData
 
+-- TODO [short links] StrEncoding instances?
+data InvitationLinkData = InvitationLinkData
+  { profile :: Profile
+  }
+  deriving (Show)
+
+data AddressLinkData = AddressLinkData
+  { profile :: Profile,
+    welcomeMessage :: Maybe Text
+  }
```

Comment:
For now we could merge these types. The argument not to merge would be the flag to prohibit/allow connection request messages that would only be needed for addresses. "Allow" would fit with legacy clients that all prohibit connection request messages by default, and we could enable for existing addresses via migration.